sympy/solvers/decompogen.py

```diff
@@ -28,8 +29,35 @@ def decompogen(f, symbol):
     [sin(x), sqrt(x), cos(x), x**2 + 1]
     >>> decompogen(x**4 + 2*x**3 - x - 1, x)
     [x**2 - x - 1, x**2 + x]
+    >>> decompogen(Lambda(x, sin(cos(x))), x)
+    [sin(x), cos(x)]
+    >>> decompogen(lambda x : sin(cos(x)), x)
+    [sin(x), cos(x), x]
 
     """
+    from sympy.core import Lambda
+    from sympy.geometry.util import _uniquely_named_symbol
+    from sympy.core.function import FunctionClass
+    from sympy.utilities.misc import func_name
+    from sympy.core.symbol import Symbol, Dummy
+
+    if len(args) is not 2:
+        raise ValueError('decompogen expects 2 args (function, symbol), got: %s' % len(args))
```

Comment:
This is rather a digression from the previous api, earlier the fact that two arguments are required was explicit, now it is implicit.